robottelo/hosts.py

```diff
@@ -1551,17 +1551,18 @@ def is_remote_db(self):
         )
 
     def capsule_certs_generate(self, capsule, cert_path=None, **extra_kwargs):
-        """Generate capsule certs, returning the cert path and the installer command args"""
-        command = InstallerCommand(
-            command='capsule-certs-generate',
-            foreman_proxy_fqdn=capsule.hostname,
-            certs_tar=cert_path or f'/root/{capsule.hostname}-certs.tar',
-            **extra_kwargs,
+        """Generate capsule certs, returning the cert path, installer command stdout and args"""
+        cert_file_path = cert_path or f'/root/{capsule.hostname}-certs.tar'
+        result = self.install(
+            InstallerCommand(
+                command='capsule-certs-generate',
+                foreman_proxy_fqdn=capsule.hostname,
+                certs_tar=cert_file_path,
+                **extra_kwargs,
+            )
         )
-        result = self.execute(command.get_command())
         install_cmd = InstallerCommand.from_cmd_str(cmd_str=result.stdout)
-        install_cmd.opts['certs-tar-file'] = f'/root/{capsule.hostname}-certs.tar'
-        return install_cmd
+        return cert_file_path, result, install_cmd
```

Comment:
It seems like every time you run `capsule_certs_generate` you don't actually need all three of these values. Maybe you could just return them as a dictionary and then call the keys when you need them.

Reply:
yes, its possible, but calling it with keys or unpack a tuple, I don't see any difference